File: code/community/1301688561354936322/arabic-language-support-deepgram-tts.cs
using System;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json;

class Program
{
    static async Task Main(string[] args)
    {
        var apiKey = Environment.GetEnvironmentVariable("DEEPGRAM_API_KEY");
        var url = "https://api.deepgram.com/v1/speak";

        using var client = new HttpClient();

        var content = new StringContent(JsonConvert.SerializeObject(new
        {
            text = "Hello, world!",
            voice = "en-US"
        }), Encoding.UTF8, "application/json");

        client.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Bearer", apiKey);

        HttpResponseMessage response = await client.PostAsync(url, content);

        if (response.IsSuccessStatusCode)
        {
            Console.WriteLine("Text-to-Speech executed successfully");
        }
        else
        {
            Console.WriteLine("Failed to execute Text-to-Speech");
        }
    }
}

File: code/community/1301688561354936322/arabic-language-support-deepgram-tts.js
require('dotenv').config();
const fetch = require('node-fetch');

const DEEPGRAM_API_KEY = process.env.DEEPGRAM_API_KEY;
const url = "https://api.deepgram.com/v1/speak";

const textToSpeak = "Hello, world!";

const headers = {
    "Authorization": `Bearer ${DEEPGRAM_API_KEY}`,
    "Content-Type": "application/json"
};

const body = JSON.stringify({
    text: textToSpeak,
    voice: "en-US"
});

fetch(url, {
    method: 'POST',
    headers: headers,
    body: body
})
.then(response => {
    if (response.ok) {
        console.log("Text-to-Speech executed successfully");
    } else {
        console.log("Failed to execute Text-to-Speech");
    }
})
.catch(error => console.error('Error:', error));